mpmath/ctx_mp_python.py

```diff
@@ -775,6 +775,8 @@ def default(ctx):
         ctx.trap_complex = False
 
     def _set_prec(ctx, n):
+        if not ctx.isfinite(n):
+            return
         ctx._prec = ctx._prec_rounding[0] = max(1, int(n))
```

Comment:
This seems like an odd fix: just ignore the precision if it is `mpmath.inf`?

Is it valid to set the `ctx.prec = mpmath.inf`?

If it is valid then surely it should be used for something rather than ignored. If it is not valid then surely it should raise an error rather than being ignored. Either way ignoring does not seem like the right thing to do.

I would have thought that the right way to fix this is here by checking if `M` is infinite before doing `ctx.prec += M`:
https://github.com/mpmath/mpmath/blob/dcc591cb678dcef0c0b37f45d430d105ee258766/mpmath/functions/bessel.py#L159-L172

Reply:
> This seems like an odd fix: just ignore the precision if it is mpmath.inf?

Yes, but see: https://github.com/mpmath/mpmath/pull/792#issuecomment-2071327300

> Is it valid to set the ctx.prec = mpmath.inf?

Yes, that was legal before.

> I would have thought that the right way to fix this is here by checking if M is infinite before doing ctx.prec += M:

That was my original fix.  The problem is that this will be required for many functions, that rely on hypercomb().  While with this approach we just fall deeper and rely on this function to handle special values for both mp and fp context.

BTW, to workaround SymPy failure we could use ctx.isinf instead.